src/Parser.hs

```diff
@@ -5,113 +5,112 @@ module Parser
       ) where
 
 import Chord
-import Common.Utils
+import Common.Utils (rightToMaybe)
 
-import Text.Parsec (option, optionMaybe, (<|>), parse, many, eof)
-import Text.Parsec.String (Parser)
-import Text.ParserCombinators.Parsec.Combinator (many1)
+import Text.Parsec
+    ( choice
+    , option
+    , optionMaybe
+    , (<|>)
+    , (<?>)
+    , parse
+    , many
+    , many1
+    , unexpected
+    , eof )
+import Text.Parsec.String (Parser) -- TODO: We're gonna want to do better than
+                                   --       this... Fortunately that won't
+                                   --       break anything, if we're careful.
 import Text.Parsec.Char (string, char, oneOf, digit)
 import Text.Parsec.Prim (try)
 
 import Data.Maybe (fromMaybe)
 
+-- TODO: Let's be a little more robust here
 parseChord :: String -> Maybe Chord
-parseChord s =
-  case parse parserChord "" s of
-    Left err  -> Nothing
-    Right xs  -> Just xs
-
+parseChord s = rightToMaybe $ parse parserChord "" s
 
+-- N.B. This should only be used when it's absolutely necessary
```

Comment:
Using `try` is expensive; it essentially forces the parser to 'rewind' its state. Rewinding like this should be avoided where possible, hence the note.